libavpipe/src/avpipe_xc.c

```diff
@@ -4333,6 +4333,26 @@ avpipe_probe(
             stream_probes_ptr->codec_type = decoder_ctx.format_context->streams[i]->codecpar->codec_type;
         }
         stream_probes_ptr->codec_name[MAX_CODEC_NAME] = '\0';
+
+        // Estimate duration if not provided by the stream format
+        if (s->duration <= 0) {
```

Comment:
Maybe modify this check to something like

```
if (s->duration == 0 || s->duration == AV_NOPTS_VALUE)
```

because I believe those are the only two cases? But any negative value isn't super meaningful, so I don't think this is a necessary change at all.

Reply:
I used duration <=0 because the duration will be provided by particular format implementations and I don't think it's guaranteed to be 0 or NOPTS in all cases.  We don't want to return a duration that's negative, so I would have to do something else about negative values that are not NOPTS - so this code is more compact and addresses both the expected and the unexpected values.